src/main/java/net/sf/mzmine/modules/peaklistmethods/alignment/adap3/ADAP3AlignerTask.java

```diff
@@ -0,0 +1,419 @@
+                PeakListRow row = findRow(peakLists,
+                        component.getSampleID(i),
+                        component.getComponent(i).getBestPeak().getInfo().peakID);
+
+                if (row != refRow)
+                    newRow.addPeak(row.getRawDataFiles()[0], row.getBestPeak());
+            }
+
+
+            PeakIdentity identity = refRow.getPreferredPeakIdentity();
+
+            if (identity != null)
+                newRow.addPeakIdentity(identity, true);
+
+            newRow.setComment("Alignment Score = " + Double.toString(component.getScore()));
+
+            // -----------------------------------------------
+            // Determine the quantitative mass and intensities
+            // -----------------------------------------------
+
+            double mass = getQuantitativeMass(component);
+            double mzTolerance = parameters.getParameter(ADAP3AlignerParameters.MZ_RANGE)
+                    .getValue()
+                    .getMzTolerance();
+
+            SimplePeakInformation information = new SimplePeakInformation();
+            information.addProperty("REFERENCE FILE", refRow.getRawDataFiles()[0].getName());
+            information.addProperty("QUANTITATION MASS", Double.toString(mass));
+
+            List<Component> components =
+                    new ArrayList<>(component.getComponents());
+
+            for (int i = 0; i < components.size(); ++i) {
+                Component c = components.get(i);
+                RawDataFile file = findFile(peakLists, component.getSampleID(i));
+
+                double minDistance = Double.MAX_VALUE;
+                double intensity = 0.0;
+
+                for (Entry<Double, Double> e : c.getSpectrum().entrySet()) {
+                    double mz = e.getKey();
+                    double distance = Math.abs(mz - mass);
+
+                    if (distance > mzTolerance) continue;
+
+                    if (distance < minDistance) {
+                        minDistance = distance;
+                        intensity = e.getValue();
+                    }
+                }
+
+                information.addProperty(
+                        "QUANTITATION INTENSITY for " + file.getName(),
+                        Double.toString(intensity));
+            }
+
+            newRow.setPeakInformation(information);
+
+            alignedPeakList.addRow(newRow);
+        }
+
+        return alignedPeakList;
+    }
+
+    private Component getComponent(final PeakListRow row) {
+        final PeakIdentity identity = row.getPreferredPeakIdentity();
+
+        if (row.getNumberOfPeaks() == 0)
+            throw new IllegalArgumentException("No peaks found");
+
+        // Read Spectrum information        
+        NavigableMap<Double, Double> spectrum = new TreeMap<>();
+
+        IsotopePattern pattern = row.getBestIsotopePattern();
+
+        if (pattern == null)
+            throw new IllegalArgumentException("ADAP Alignment requires mass "
```

Comment:
Throwing an IllegalArgumentException does not seem correct here - you should just set the error message and exit the method gracefully.